snap/snapcraft.yaml

```diff
@@ -17,31 +33,49 @@ apps:
       - wayland
     plugs:
       - opengl
-    environment:
-      XKB_CONFIG_ROOT: $SNAP/usr/share/X11/xkb
-      XDG_CONFIG_HOME: $SNAP_DATA
-      XCURSOR_PATH:    $SNAP/icons
-      __EGL_VENDOR_LIBRARY_DIRS: $SNAP/etc/glvnd/egl_vendor.d:$SNAP/usr/share/glvnd/egl_vendor.d
 
 parts:
+  ppa-mir-release:
+    plugin: nil
+    override-pull: |
+      sudo apt --assume-yes install software-properties-common
+      sudo add-apt-repository -yu ppa:mir-team/release
```

Comment:
I'm quite uneasy about this approach... There's no guarantee this even works, if `snapcraft` decides to cache all dependencies prior to any of the `pull` steps.

IIUC this "solves" building outside of Launchpad builders. But they are our main consumer of this, and we can instead document that one of the Mir PPAs is required for this snap to build.

Reply:
This approach has been documented multiple times on the snapcraft forums. Guaranteed or not, there are a lot of snaps that depend on it.

Reply:
FWIW [snapcraft local plugins](https://docs.snapcraft.io/writing-local-plugins/5125) would be a better approach here, but as we're thinking we'd rather have a `miral-framework` instead, I'm OK with this for now.